x/group/keeper/keeper.go

```diff
@@ -230,3 +240,59 @@ func (k Keeper) MaxMetadataLength() uint64 { return k.config.MaxMetadataLen }
 func (k Keeper) GetGroupSequence(ctx sdk.Context) uint64 {
 	return k.groupTable.Sequence().CurVal(ctx.KVStore(k.key))
 }
+
+func (k Keeper) IterateVPEndProposals(ctx sdk.Context, cb func(proposal group.Proposal) (stop bool)) error {
+	timeBytes := sdk.FormatTimeBytes(ctx.BlockTime())
+	it, err := k.ProposalsByVotingPeriodEnd.Get(ctx.KVStore(k.key), sdk.PrefixEndBytes(timeBytes))
```

Comment:
use `PrefixScan` instead to get all proposals with VotingPeriodEnd before block time:

```suggestion
	it, err := k.ProposalsByVotingPeriodEnd.PrefixScan(ctx.KVStore(k.key), nil, timeBytes)
```

Reply:
Thanks for the suggestion :pray: